ReactWindows/ReactNative.Shared/Reflection/EnumHelpers.cs

```diff
@@ -39,6 +32,22 @@ public static T Parse<T>(string value)
             return Parse<T>(value);
         }
 
+        public static bool TryParse<T>(string value, out T result)
+        {
+            var lookup = s_enumCache.GetOrAdd(
+                typeof(T),
+                type => EnumToDictionary(type));
+
+            if (!lookup.TryGetValue(Normalize(value), out var objectResult))
+            {
+                result = default(T);
+                return false;
+            }
+
+            result = (T)objectResult;
```

Comment:
Is this guaranteed to be T? maybe it is better to use 'as'. I think (not sure) that (T) throws exception if objectResult is not T

Reply:
yes, if it's not T then s_enumCache contains invalid data and we must throw. Also it's existing code - it was like this before - see the Parse method above. The "Try" part of this method is if the value is not there in the cache. not if there is a value but of wrong type.

---
In reply to: [168846411](https://github.com/gmangov/react-native-windows/pull/1#discussion_r168846411) [](ancestors = 168846411)